### ISSUE TITLE
Localization support

### ISSUE THREAD

**Issue by jermanuts** [BKSalman/ytdlp-gui#26]
Make it possible to translate the app in different languages. Hope RTL is supported too for Arabic etc.

**Comment by BKSalman** [comment]
Good idea, I will think of a way to create the base for this and make it easy for people to submit their translations

If you have suggestions feel free to share them

**Comment by jermanuts** [comment]
> Good idea, I will think of a way to create the base for this and make it easy for people to submit their translations

[Weblate](https://weblate.org/en/)

>Also RTL should indeed be supported by iced already

You sure? https://github.com/iced-rs/iced/issues/250

**Comment by BKSalman** [comment]
Since this feature is not a priority, it will be delayed, unless someone wants to add the ground work in the code

**Comment by BKSalman** [comment]
The way cosmic does localization could be useful, but I still didn't try it.

Will try to look at it later inshallah

**Comment by BKSalman** [comment]
079e1e8 adds the base for i18n, it's still very basic, but it should help people add their own translations

tho Arabic/RTL still isn't [officially supported in iced](https://github.com/iced-rs/iced/issues/1981#issuecomment-1657218038)

**Comment by dcampbell24** [comment]
I translated https://github.com/dcampbell24/hnefatafl/blob/main/examples/hnefatafl-client.rs into a bunch of languages including Arabic.

**Comment by BKSalman** [comment]
Hmm, looks like it got fixed judging by the latest comments in this [issue](https://github.com/iced-rs/iced/issues/1981), I will test the git version, but I think I will wait for the 0.14 release to switch to it and release a new ytdlp-gui version
